app/services/course/assessment/question/programming_import_service.rb

```diff
@@ -125,4 +126,9 @@ def infer_test_case_type(test_case_name)
   def parse_test_report(test_report)
     Course::Assessment::ProgrammingTestCaseReport.new(test_report).test_cases
   end
+
+  # Auto grade all answers for the question which are not in the attempting state.
+  def regrade_answers
+    @question.answers.select { |ans| ans.workflow_state != 'attempting' }.map(&:auto_grade!)
```

Comment:
Use SQL query instead of `select`, there could be thousands of answers. Also should use `find_each` to fetch them in batches.